crates/sc-consensus-subspace/src/slot_worker.rs

```diff
@@ -50,6 +50,31 @@ use subspace_verification::{
     VerifySolutionParams,
 };
 
+#[derive(Clone)]
+pub(super) struct SlotWorkerSyncOracle<SO>
+where
+    SO: SyncOracle + Send + Sync + Clone,
+{
+    pub(super) force_authoring: bool,
+    pub(super) inner: SO,
+}
+
+impl<SO> SyncOracle for SlotWorkerSyncOracle<SO>
+where
+    SO: SyncOracle + Send + Sync + Clone,
+{
+    fn is_major_syncing(&self) -> bool {
+        // This allows slot worker to produce blocks even when it is offline, which according to
+        // modified Substrate fork will happen when node is offline or connected to non-synced peers
+        // (default state)
+        self.inner.is_major_syncing() && !self.force_authoring
```

Comment:
Just a minor thing, but it might be better to short circuit this condition:
```rust
!self.force_authoring && self.inner.is_major_syncing()
```

Reply:
It is just an atomic read in the method, but sure, applied.